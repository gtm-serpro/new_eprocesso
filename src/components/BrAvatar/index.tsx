import Icon from "@/common/Icon";
import classNames from "classnames";

type AvatarType = "image" | "icon" | "letter";

export interface BrAvatarProps {
  className?: string;
  title?: string;
  src?: string;
  type?: AvatarType;
  size?: "medium" | "large";
}

const SwitchType = ({ type, src }: { type?: AvatarType; src: string }) => {
  switch (type) {
    case "image":
      return <img src={src} alt="Avatar" />;
    case "icon":
      return <Icon icon={src} aria-hidden="true" />;
    case "letter":
      return <>{src}</>;
    default:
      return <Icon icon="user" aria-hidden="true" />;
  }
};

const BrAvatar = (props: BrAvatarProps) => {
  const { className, title, src = "", type, size } = props;

  const rootClasses = classNames("br-avatar", title, size, className);
  const innerClasses = classNames({
    image: !type || type === "image" || type === "icon",
    letter: type === "letter",
  });

  return (
    <span className={rootClasses} title={title}>
      <span className={innerClasses}>
        <SwitchType type={type} src={src} />
      </span>
    </span>
  );
};

export default BrAvatar;
