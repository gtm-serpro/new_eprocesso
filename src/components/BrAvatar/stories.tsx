import { Meta, Story } from "@storybook/react";

import BrAvatar, { BrAvatarProps } from ".";

export default {
  title: "Utilidades/Avatar",
  component: BrAvatar,
} as Meta;

export const Simples: Story<BrAvatarProps> = (args) => <BrAvatar {...args} />;

export const Fotografico: Story<BrAvatarProps> = (args) => <BrAvatar {...args} />;
Fotografico.args = {
  src: "https://picsum.photos/id/823/400",
  type: "image",
};
