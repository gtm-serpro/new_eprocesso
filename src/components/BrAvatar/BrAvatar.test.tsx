import { render, screen } from "@testing-library/react";

import BrAvatar from ".";

describe("<BrAvatar />", () => {
  it("deveria renderizar o avatar simples e com ícone", () => {
    const { container, getByTitle } = render(<BrAvatar title="alt do avatar" />);
    getByTitle("alt do avatar");
    expect(container.firstChild).toHaveClass("br-avatar");
    expect(container.firstChild).toMatchSnapshot();
  });

  it("deveria renderizar o avatar simples e com imagem", () => {
    const { container, getByTitle } = render(<BrAvatar src="imagem" title="alt do avatar" />);
    getByTitle("alt do avatar");
    expect(container.firstChild).toHaveClass("br-avatar");
  });

  it("deveria renderizar o avatar large se prop informada", () => {
    const { container } = render(<BrAvatar size="large" />);
    expect(container.firstChild).toHaveClass("large");
  });

  it("deveria renderizar o avatar medium se prop informada", () => {
    const { container } = render(<BrAvatar size="medium" />);
    expect(container.firstChild).toHaveClass("medium");
  });

  it("deveria renderizar o título se informado", () => {
    render(<BrAvatar title="Fulana de Tal" />);
    screen.getByTitle("Fulana de Tal");
  });
});
