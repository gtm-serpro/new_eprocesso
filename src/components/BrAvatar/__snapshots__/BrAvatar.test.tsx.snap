// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BrAvatar /> deveria renderizar o avatar simples e com Ã­cone 1`] = `
<span
  class="br-avatar alt do avatar"
  title="alt do avatar"
>
  <span
    class="image"
  >
    <i
      aria-hidden="true"
      class="fas fa-user"
    />
  </span>
</span>
`;
